Counter/src/app/hall-of-fame/hall-of-fame.component.ts

```diff
@@ -0,0 +1,42 @@
+import { Component, OnInit } from '@angular/core';
+import { Observable } from 'rxjs';
+import { HallOfFameService } from '../shared/hall-of-fame-service';
+import { Pokemon } from '../shared/pokemon';
+
+@Component({
+  selector: 'app-hall-of-fame',
+  templateUrl: './hall-of-fame.component.html',
+  styleUrls: ['./hall-of-fame.component.css']
+})
+export class HallOfFameComponent implements OnInit {
+  shiny!: Observable<Pokemon[]>;
+  isTime: Boolean = true;
+  isAsc: Boolean = false;
+  isDesc: Boolean = false;
+
+  constructor(private hallOfFameService: HallOfFameService) { }
+
+  ngOnInit(): void {
+    this.getShiny();
+  }
+  
+  getShiny() {
+    this.shiny = this.hallOfFameService.getShiny();
+    this.isTime = true;
+    this.isAsc = false;
+    this.isDesc = false;
+  }
+  sortAsc() {
```

Comment:
Instead of recreating the observable every time the user clicks a button, I'd like you to try a different pattern.

Create a Subject or BehaviorSubject to control any changes to the original observable (the Observable<Pokemon[]>) Then when you sort all you'll need to do is `next(...)`. When you do that next, it should trigger reloading the pokemon observable then map or sort or filter or anything at all with that pokemon observable.

This is a common "refresh" pattern for sorting, filtering, refreshing or manipulating a list many times over. Then you can subscribe to just one observable and that observable will emit changes when it's relevant to do so. Don't map the Subject back to the view though. You don't want the view to be able to control your observable. use `as Observable` to expose it to the view.

I have some examples of this if you get stuck.

Reply:
I am stuck on how to start implementing this; do you mind sharing some examples?